mobile/lib/features/stable/stable_confirmation_sheet.dart

```diff
@@ -0,0 +1,266 @@
+              return StableDialog(
+                pendingOrder: pendingOrder!,
+              );
+            });
+      });
+    }
+
+    final now = DateTime.now();
+
+    return Form(
+        key: _formKey,
+        child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              FutureBuilder<(ChannelInfo?, Amount)>(
+                  future: _getChannelInfo(channelInfoService),
+                  // a previously-obtained Future<String> or null
+                  builder: (BuildContext context, AsyncSnapshot<(ChannelInfo?, Amount)> snapshot) {
+                    if (!snapshot.hasData) {
+                      return Container();
+                    }
+
+                    var (channelInfo, lspMaxChannelCapacity) = snapshot.data!;
+                    Amount tradeFeeReserve = channelInfoService.getTradeFeeReserve();
+
+                    Amount channelCapacity = lspMaxChannelCapacity;
+
+                    Amount initialReserve = channelInfoService.getInitialReserve();
+
+                    Amount channelReserve = channelInfo?.reserve ?? initialReserve;
+                    int totalReserve = channelReserve.sats + tradeFeeReserve.sats;
+
+                    int usableBalance = max(walletInfo.balances.lightning.sats - totalReserve, 0);
+                    // the assumed balance of the counterparty based on the channel and our balance
+                    // this is needed to make sure that the counterparty can fulfil the trade
+                    int counterpartyUsableBalance = max(
+                        channelCapacity.sats - (walletInfo.balances.lightning.sats + totalReserve),
+                        0);
+
+                    return Container(
+                      padding: const EdgeInsets.all(20),
+                      child: Column(
+                        children: [
+                          const Text(
+                            "How much do you want to stabilize?",
+                            style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
+                          ),
+                          const SizedBox(height: 20),
+                          Row(
+                            crossAxisAlignment: CrossAxisAlignment.start,
+                            children: [
+                              Flexible(
+                                  child: Selector<StableValuesChangeNotifier, double>(
+                                      selector: (_, provider) =>
+                                          provider.stableValues().quantity ?? 0.0,
+                                      builder: (context, quantity, child) {
+                                        return FiatAmountInputField(
+                                          value: tradeValues.quantity!,
+                                          hint: "e.g. 100 USD",
```

Comment:
🙃 might be a while before we allow $100 contracts 😅
```suggestion
hint: "e.g. 10 USD",
```